File: app/src/pfp/java/social/entourage/android/map/filter/MapFilter.java
package social.entourage.android.map.filter;

import java.io.Serializable;

/**
 * Created by Mihai Ionescu on 04/07/2018.
 */
public class MapFilter implements MapFilterInterface, Serializable {

    private static final long serialVersionUID = 1562838744560618668L;

    // ----------------------------------
    // Attributes
    // ----------------------------------

    public boolean entourageTypeNeighborhood = true;
    public boolean entourageTypePrivateCircle = true;
    public boolean entourageTypeOuting = true;
    public boolean includePastEvents = false;

    // ----------------------------------
    // Lifecycle
    // ----------------------------------

    private static MapFilter ourInstance = new MapFilter();

    public static MapFilter getInstance() {
        return ourInstance;
    }

    // ----------------------------------
    // MapFilterInterface implementation
    // ----------------------------------

    @Override
    public String getTypes() {
        StringBuilder entourageTypes = new StringBuilder("");

        if (entourageTypeNeighborhood) {
            entourageTypes.append("nh");
        }
        if (entourageTypePrivateCircle) {
            if (entourageTypes.length() > 0) entourageTypes.append(",");
            entourageTypes.append("pc");
        }
        if (entourageTypeOuting) {
            if (entourageTypes.length() > 0) entourageTypes.append(",");
            entourageTypes.append("ou");
        }

        return entourageTypes.toString();
    }

    @Override
    public boolean onlyMyEntourages() {
        return false;
    }

    @Override
    public int getTimeFrame() {
        return 720; // 30 days
    }

    @Override
    public boolean onlyMyPartnerEntourages() {
        return false;
    }

    @Override
    public void entourageCreated() {

    }

    @Override
    public void validateCategories() {

    }
}

File: app/src/pfp/java/social/entourage/android/map/filter/MapFilter.kt
package social.entourage.android.map.filter

import java.io.Serializable

/**
 * Created by Mihai Ionescu on 04/07/2018.
 */
class MapFilter : MapFilterInterface, Serializable {
    // ----------------------------------
    // Attributes
    // ----------------------------------
    @JvmField
    var entourageTypeNeighborhood = true
    @JvmField
    var entourageTypePrivateCircle = true
    @JvmField
    var entourageTypeOuting = true
    @JvmField
    var includePastEvents = false

    // ----------------------------------
    // MapFilterInterface implementation
    // ----------------------------------
    override fun getTypes(): String? {
        val entourageTypes = StringBuilder("")
        if (entourageTypeNeighborhood) {
            entourageTypes.append("nh")
        }
        if (entourageTypePrivateCircle) {
            if (entourageTypes.length > 0) entourageTypes.append(",")
            entourageTypes.append("pc")
        }
        if (entourageTypeOuting) {
            if (entourageTypes.length > 0) entourageTypes.append(",")
            entourageTypes.append("ou")
        }
        return entourageTypes.toString()
    }

    override fun getTimeFrame(): Int {
        return 720 // 30 days
    }

    override fun showPastEvents(): Boolean {
        return includePastEvents
    }

    override fun entourageCreated() {}
    override fun validateCategories() {}
    override fun isDefaultFilter(): Boolean {
        //TODO
        return when {
            !entourageTypeNeighborhood -> true
            !entourageTypePrivateCircle -> true
            !entourageTypeOuting -> true
            else -> false
        }
    }

    override fun setDefaultValues(isProUser: Boolean) {
        entourageTypeNeighborhood = true
        entourageTypePrivateCircle = true
        entourageTypeOuting = true
        includePastEvents = false
    }

    companion object {
        private const val serialVersionUID = 1562838744560618668L

        // ----------------------------------
        // Lifecycle
        // ----------------------------------
        val instance = MapFilter()
    }
}